udf-compiler/src/main/scala/com/nvidia/spark/udf/Plugin.scala

```diff
@@ -50,7 +52,12 @@ case class LogicalPlanRules() extends Rule[LogicalPlan] with Logging {
     // iterating over NamedExpression
     exp match {
       case f: ScalaUDF => // found a ScalaUDF
-        GpuScalaUDFLogical(f).compile(conf.isTestEnabled)
+        // If it implements the RapidsUDF interface, no need to compile it.
+        if (!getRapidsUDFInstance(f.function).isEmpty) {
+          exp
```

Comment:
If we're not going to compile the UDF then the expression should be treated like any other expression and recurse, in case one of its input expressions (which could be yet another UDF) could be compiled, e.g.:
```scala
  case f: ScalaUDF if getRapidsUDFInstance(f.function).isDefined {
    GpuScalaUDFLogical(f).compile(conf.isTestEnabled)
  case _ =>
....
```

Reply:
sorry, missed this. Updated.